Straight_skeleton_2/doc/Straight_skeleton_2/CGAL/extrude_skeleton.h

```diff
@@ -0,0 +1,94 @@
+namespace CGAL {
+
+/**
+* \ingroup PkgStraightSkeleton2Extrusion
+*
+* \brief constructs the straight skeleton-based extrusion of a polygon with holes.
+*
+* Given a polygon with holes and a set of weights, the skeleton extrusion is a volume constructed
+* from the weighted straight skeleton by associating a height to the vertices of the skeleton,
+* which corresponds to the time at the vertex. The input polygon is placed at `z = 0`.
+*
+* This function allows cropping the extruded skeleton at a maximum height, using the optional
+* `maximum_height` named parameter.
+*
+* The result is a closed, 2-manifold surface triangle mesh. Note that this mesh can have non-local
+* self-intersections in some case of cropping.
+*
+* @tparam PolygonWithHoles must be a model of `SequenceContainer` with value type `InK::Point_2` (e.g. `Polygon_2<InK>`)
+                           or a model of `GeneralPolygonWithHoles_2` (e.g. `Polygon_with_holes_2<InK>`).
+* @tparam PolygonMesh a model of `MutableFaceGraph`
+* @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+*
+* @param polygon the polygon with holes
+* @param out the output polygon mesh
+* @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+*
+* \cgalNamedParamsBegin
+*   \cgalParamNBegin{weights}
```

Comment:
anything about references taken on the range?

Reply:
Comment not addressed

Reply:
It's taken by const& and a copy is immediately taken since I modify the weights and angles. I don't really see the point of explaining that to a user, what would you like to see written?

Reply:
If the object is taken by reference it must be said IMO.